Ease using `make test` with Docker and improve related documentation (#3194)
os-autoinst/openQA: docs/Contributing.asciidoc

changes are performed, e.g. different OS version or packages from forked OBS
project, etc.

[[circleci-local-docker]]
=== Run tests locally using Docker

> One way is to build an image using `build_local_docker.sh` script, start a

Reviewer: ```suggestion
One way is to build an image using the `build_local_docker.sh` script, start a
```

container and then use the same commands one would use to test locally.

Pull the latest base image (otherwise it may be outdated):
```
docker pull registry.opensuse.org/devel/openqa/ci/containers/base:latest
```

> Create an image called `localtest` based on the content of `dependnencies.txt`

Reviewer: ```suggestion
Create an image called `localtest` based on the contents of `dependencies.txt`
```

and `autoinst`:
```
.circleci/build_local_docker.sh
```

Mount the openQA checkout under `/opt/testing_area` within the container and run
tests as usual, e.g.:
```
docker run -it --rm -v $PWD:/opt/testing_area localtest bash -c 'make test TESTS=t/ui/25*'
```

Alternatively, start the container and execute commands via `docker exec`, e.g.:
```
docker run --rm --name t1 -v $PWD:/opt/testing_area localtest tail -f /dev/null & sleep 1
docker exec -it t1 bash -c 'make test TESTS=t/ui/25-developer_mode.t'
docker stop -t 0 t1
```

=== Run tests using the circleci tool

> After installing the `circleci` tool following commands will be available.

Reviewer: ```suggestion
After installing the `circleci` tool the following commands will be available.
```

=== Changing config.cnf

> Command to verify yaml with the `circleci` tool:

Reviewer: ```suggestion
Command to verify the YAML with the `circleci` tool:
```
